pkg/autoscaler/autoscaler/pdplan_autoscaler.go

```diff
@@ -31,7 +31,29 @@ import (
 
 const groupLabelKey = "group"
 
-func (am *autoScalerManager) syncPlans(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, plans []pdapi.Plan) error {
+func (am *autoScalerManager) getAutoScaledClusters(tac *v1alpha1.TidbClusterAutoScaler, components []v1alpha1.MemberType) (tcList []*v1alpha1.TidbCluster, err error) {
+	componentStrings := make([]string, len(components))
+	for _, component := range components {
+		componentStrings = append(componentStrings, component.String())
+	}
+
+	// Filter all autoscaled TidbClusters
+	requirement, err := labels.NewRequirement(label.AutoInstanceLabelKey, selection.Equals, []string{tac.Name})
+	if err != nil {
+		return
+	}
+
+	componentRequirement, err := labels.NewRequirement(label.AutoComponentLabelKey, selection.In, componentStrings)
+	if err != nil {
+		return
+	}
+
+	selector := labels.NewSelector().Add(*requirement).Add(*componentRequirement)
+	tcList, err = am.tcLister.TidbClusters(tac.Spec.Cluster.Namespace).List(selector)
```

Comment:
`tac.Spec.Cluster.Namespace` seems not defaulted.

Reply:
it is defaulted in https://github.com/pingcap/tidb-operator/blob/cf85ab3b4be63e089a4088ca083494b0cdbb67cb/pkg/autoscaler/autoscaler/autoscaler_manager.go#L78